src/main/scala/io/iohk/ethereum/jsonrpc/ExpiringMap.scala

```diff
@@ -0,0 +1,69 @@
+package io.iohk.ethereum.jsonrpc
+
+import java.time.temporal.ChronoUnit
+import java.time.Duration
+
+import io.iohk.ethereum.jsonrpc.ExpiringMap.ValueWithDuration
+
+import scala.collection.mutable
+import scala.util.Try
+
+object ExpiringMap {
+
+ case class ValueWithDuration[V](value: V, expiration: Duration)
+
+  def empty[K, V](defaultElementRetentionTime: Duration): ExpiringMap[K, V] =
+    new ExpiringMap(mutable.Map.empty, defaultElementRetentionTime)
+}
+/**
+  * Simple wrapper around mutable map which enriches each element with expiration time (specified by user or default)
+  * Map is passive which means it only check for expiration and remove expired element during get function.
+  * Duration in all calls is relative to current System.nanoTime()
+  */
+//TODO: Make class thread safe
+class ExpiringMap[K, V] private (val underLaying: mutable.Map[K, ValueWithDuration[V]],
+                                 val defaultRetentionTime: Duration) {
+  private val maxHoldDuration = ChronoUnit.CENTURIES.getDuration
```

Comment:
Is this to represent _forever_? `Duration.Inf`?

Reply:
Oh, I thought we were using `scala.concurrent.duration.Duration` rather than `java.time.Duration`. I guess that's ok, but is there any reason for this choice?

Reply:
`java.time` is enough for our simple use case, and we are avoiding the middleman. (some implicit conversions beneath, additional object creation etc.)
For an external user of class, i think 100 years is enough to think of it as forever.
Also as docs states `This class is not meant as a general purpose representation of time, it is optimized for the needs of scala.concurrent.` and we are not using any `scala.concurrent` here.